realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -1520,6 +1630,19 @@ public void endsWith_nullForNullableStrings() {
                 (String) null).findFirst().getFieldStringNotNull());
     }
 
+    // Querying nullable field with like - nulls do not match either '?' or '*'
+    @Test
+    public void like_nullForNullableStrings() {
+        TestHelper.populateTestRealmForNullTests(realm);
+        RealmResults<NullTypes> resultList = realm.where(NullTypes.class)
+                .like(NullTypes.FIELD_STRING_NULL, "*").findAll();
+        assertEquals(2, resultList.size());
```

Comment:
I thought all 3 string row have `null` value, but it is not.  So this is correct 👍 
```
// +-+--------+------+---------+--------+--------------------+
// | | string | link | numeric | binary | numeric (not null) |
// +-+--------+------+---------+--------+--------------------+
// |0| Fish   |    0 |       1 |    {0} |                  1 |
// |1| null   | null |    null |   null |                  0 |
// |2| Horse  |    1 |       3 |  {1,2} |                  3 |
// +-+--------+------+---------+--------+--------------------+
```